### ISSUE TITLE
Not compatible with Magento 2.4.7

### ISSUE THREAD

**Issue by verhaeghep** [magento/meta-for-magento2#62]
### Preconditions (*)

1."magento/product-community-edition": "2.4.7",
2."meta/meta-for-magento2": "*"

### Steps to reproduce (*)

1. Try to require the package for magento 2.4.7
2. Fail doing so because the package restricted to 2.4.6-p5

### Expected result (*)

1. Composer install runs successfully

### Actual result (*)

```
    - magento/product-community-edition 2.4.7-beta2 requires magento/framework-message-queue 100.4.7-beta2 -> satisfiable by magento/framework-message-queue[100.4.7-beta2].
    - Conclusion: don't install magento/framework-message-queue 100.4.7-beta2 (conflict analysis result)
    - magento/product-community-edition 2.4.7-beta3 requires magento/framework-message-queue 100.4.7-beta3 -> satisfiable by magento/framework-message-queue[100.4.7-beta3].
    - Conclusion: don't install magento/framework-message-queue 100.4.7-beta3 (conflict analysis result)
    - magento/product-community-edition 2.4.7 requires magento/framework-message-queue 100.4.7 -> satisfiable by magento/framework-message-queue[100.4.7].
    - Conclusion: don't install magento/framework-message-queue 100.4.7 (conflict analysis result)
    - meta/meta-for-magento2 1.3.0 requires meta/module-business-extension 1.3.0 -> satisfiable by meta/module-business-extension[1.3.0].
    - meta/meta-for-magento2 1.2.6 requires meta/module-business-extension 1.2.6 -> satisfiable by meta/module-business-extension[1.2.6].
    - meta/meta-for-magento2[1.2.0, ..., 1.2.2] require meta/module-business-extension 1.2.* -> satisfiable by meta/module-business-extension[1.2.0, ..., 1.2.6].
    - meta/meta-for-magento2 1.2.5 requires meta/module-business-extension 1.2.5 -> satisfiable by meta/module-business-extension[1.2.5].
    - meta/meta-for-magento2 1.2.4 requires meta/module-business-extension 1.2.4 -> satisfiable by meta/module-business-extension[1.2.4].
    - meta/meta-for-magento2 1.2.3 requires meta/module-business-extension 1.2.3 -> satisfiable by meta/module-business-extension[1.2.3].
    - meta/module-business-extension[1.2.2, ..., 1.3.0] require magento/framework-message-queue >=100.4.2 <100.4.7 -> satisfiable by magento/framework-message-queue[100.4.2, ..., 100.4.6].
    - meta/module-business-extension[1.2.0, ..., 1.2.1] require magento/framework-message-queue >=100.4.4 <=100.4.6 -> satisfiable by magento/framework-message-queue[100.4.4, 100.4.5, 100.4.6].
    - You can only install one version of a package, so only one of these can be installed: magento/framework-message-queue[100.0.2, ..., 100.4.7].
    - magento/product-community-edition 2.4.7-beta1 requires magento/framework-message-queue 100.4.7-beta1 -> satisfiable by magento/framework-message-queue[100.4.7-beta1].
    - Root composer.json requires magento/product-community-edition ^2.4.7 -> satisfiable by magento/product-community-edition[2.4.7-beta1, 2.4.7-beta2, 2.4.7-beta3, 2.4.7].
    - Root composer.json requires meta/meta-for-magento2 ^1.2 -> satisfiable by meta/meta-for-magento2[1.2.0, ..., 1.3.0].

**Comment by sol-loup** [comment]
Hey @verhaeghep  -- we were able to install the extension on a 2.4.7. While we encountered a similar issue to the one you describe here, it was resolvable by purging the composer.lock and reinstalling clean. The root composer.json of the meta-for-magento2 extension doe snot specify a hard version of magento/framework-message-queue -- so issues of this nature are typically caused by a composer.lock incompatibility

**Comment by sol-loup** [comment]
<img width="1305" alt="Screenshot 2024-04-22 at 4 30 56 PM" src="https://github.com/magento/meta-for-magento2/assets/129231008/015433aa-682e-4a32-acb8-360be7a11e34">

**Comment by matei** [comment]
@sol-loup it seems that:
- meta/module-sales 1.3.0 (part of meta/meta-for-magento2) requires  magento/module-tax >=100.4.2 <100.4.7
- Magento 2.4.7 has magento/module-tax at version: "100.4.7"

**Comment by lytesaber** [comment]
@sol-loup It may be correct within the Github project however the Meta for Magento 2 module packaged / mirrored on the Adobe Commerce Marketplace has different composer require version contraints. Comparing the composer.json installed via the Adobe Commerce Marketplace versus the composer.json on Github you can see there's version contraints that cause the composer install / update issue @matei is referring to.

**meta\module-sales\composer.json (Installed from Adobe Commerce Marketplace)**
```
{
    "name": "meta/module-sales",
    "description": "Meta Business Extension Sales module",
    "type": "magento2-module",
    "license": "Apache-2.0",
    "version": "1.3.0",
    "require": {
        "php": "~7.4||~8.1.0||~8.2.0",
        "magento/framework": ">=103.0.2 <103.0.7",
        "magento/module-sales": ">=103.0.2 <103.0.7",
        "magento/module-sales-sequence": ">=100.4.1 <100.4.4",
        "magento/module-backend": ">=102.0.2 <102.0.7",
        "magento/module-config": ">=101.2.2 <101.2.7",
        "magento/module-catalog": ">=104.0.2 <104.0.7",
        "magento/module-configurable-product": ">=100.4.2 <100.4.7",
        "magento/module-checkout": ">=100.4.2 <100.4.7",
        "magento/module-customer": ">=103.0.2 <103.0.7",
        "magento/module-store": ">=101.1.2 <101.1.7",
        "magento/module-directory": ">=100.4.2 <100.4.7",
        "magento/module-offline-shipping": ">=100.4.1 <100.4.6",
        "magento/module-payment": ">=100.4.2 <100.4.7",
        "magento/module-quote": ">=101.2.2 <101.2.7",
        "magento/module-sales-rule": ">=101.2.2 <101.2.7",
        "magento/module-shipping": ">=100.4.2 <100.4.7",
        "magento/module-newsletter": ">=100.4.2 <100.4.7",
        "magento/module-tax": ">=100.4.2 <100.4.7",
        "meta/module-business-extension": "1.3.0",
        "meta/module-catalog": "1.3.0"
    },
    "autoload": {
        "psr-4": {
            "Meta\\Sales\\": ""
        },
        "files": [
            "registration.php"
        ]
    }
}
``` 

#

**Comment by sol-loup** [comment]
Understood, thanks for bringing this to our attention. I'll work with our marketplace bundler team to get this resolved

**Comment by PhiIipp** [comment]
@sol-loup , magento 2.4.7  also support php 8.3 which I am using. Here I also get an error dependency not met .  Do you know if this is a bigger issue or only update of dependency checker. 

    - meta/module-business-extension[1.0.0, ..., 1.0.1] require php ~8.1.0 -> your php version (8.3.6) does not satisfy that requirement.
    - meta/module-business-extension[1.1.0, ..., 1.2.1] require php ~8.1.0||~8.2.0 -> your php version (8.3.6) does not satisfy that requirement.
    - meta/module-business-extension[1.2.2, ..., 1.3.0] require php ~7.4||~8.1.0||~8.2.0 -> your php version (8.3.6) does not satisfy that requirement.
    - Root composer.json requires meta/meta-for-magento2 * -> satisfiable by meta/meta-for-magento2[1.0.0, ..., 1.3.0].

**Comment by sol-loup** [comment]
@PhiIipp , at the moment we must take a conservative view WRT supported PHP versions due to commitments to support sellers on magento 2.4.3. We have not yet done a validation pass on PHP 8.3, but I will add it to our backlog.

**Comment by lytesaber** [comment]
@sol-loup, any update on when the module with the corrected version dependencies will be avaliable on the Adobe Commerce Marketplace?

**Comment by sol-loup** [comment]
Hey @PhiIipp , @lytesaber  -- version 1.3.1, just released to composer, updates these dependencies to <=, allowing smooth install on 2.4.7 machines:

{
    "name": "meta/module-sales",
    "description": "Meta Business Extension Sales module",
    "type": "magento2-module",
    "license": "Apache-2.0",
    "version": "1.3.1",
    "require": {
        "php": "~7.4||~8.1.0||~8.2.0",
        "magento/framework": ">=103.0.2 <=103.0.7",
        "magento/module-sales": ">=103.0.2 <=103.0.7",
        "magento/module-sales-sequence": ">=100.4.1 <=100.4.4",
        "magento/module-backend": ">=102.0.2 <=102.0.7",
        "magento/module-config": ">=101.2.2 <=101.2.7",
        "magento/module-catalog": ">=104.0.2 <=104.0.7",
        "magento/module-configurable-product": ">=100.4.2 <=100.4.7",
        "magento/module-checkout": ">=100.4.2 <=100.4.7",
        "magento/module-customer": ">=103.0.2 <=103.0.7",
        "magento/module-store": ">=101.1.2 <=101.1.7",
        "magento/module-directory": ">=100.4.2 <=100.4.7",
        "magento/module-offline-shipping": ">=100.4.1 <=100.4.6",
        "magento/module-payment": ">=100.4.2 <=100.4.7",
        "magento/module-quote": ">=101.2.2 <=101.2.7",
        "magento/module-sales-rule": ">=101.2.2 <=101.2.7",
        "magento/module-shipping": ">=100.4.2 <=100.4.7",
        "magento/module-newsletter": ">=100.4.2 <=100.4.7",
        "magento/module-tax": ">=100.4.2 <=100.4.7",
        "meta/module-business-extension": "1.3.1",
        "meta/module-catalog": "1.3.1"
    },
    "autoload": {
        "files": [
            "registration.php"
        ],
        "psr-4": {
            "Meta\\Sales\\": ""
        }
    }
}

**Comment by sol-loup** [comment]
@verhaeghep can you close if your issue has been solved?

**Comment by verhaeghep** [comment]
Issue solved since version 1.3.1
Thanks.
